app/src/test/java/com/codesoom/assignment/controllers/UserControllerTest.java

```diff
@@ -36,122 +47,238 @@
+                    })
+                            .when(userService).updateUser(eq(1L), any(UserModificationData.class));
+                }
 
-        verify(userService).deleteUser(1L);
+                @Test
+                @DisplayName("ok 와 변경된 user 를 응답한다.")
+                void It_respond_ok_and_modified_user() throws Exception {
+                    mockMvc.perform(
+                            patch("/users/1")
+                                    .contentType(MediaType.APPLICATION_JSON)
+                                    .content("{\"name\":\"TEST\",\"password\":\"test\"}")
+                                    .header("Authorization", "Bearer " + VALID_TOKEN)
+                    )
+                            .andExpect(status().isOk())
+                            .andExpect(content().string(
+                                    containsString("\"id\":1")
+                            ))
+                            .andExpect(content().string(
+                                    containsString("\"name\":\"TEST\"")
+                            ));
+
+                    verify(userService).updateUser(eq(1L), any(UserModificationData.class));
+                }
+            }
+
+            @Nested
+            @DisplayName("주어진 토큰에 있는 id 가 아닌 다른 id 를 변경하려 할 때")
+            class Context_when_not_exists_given_id_user {
+                @BeforeEach
+                void setup() {
+                    Mockito.doThrow(new AccessDeniedException("Access denied"))
+                            .when(userService).updateUser(eq(100L), any(UserModificationData.class));
+                }
+
+                @Test
+                @DisplayName("forbidden 를 응답한다.")
+                void It_respond_not_found() throws Exception {
+                    mockMvc.perform(
+                            patch("/users/100")
+                                    .contentType(MediaType.APPLICATION_JSON)
+                                    .content("{\"name\":\"TEST\",\"password\":\"TEST\"}")
+                                    .header("Authorization", "Bearer " + VALID_TOKEN)
+                    )
+                            .andExpect(status().isForbidden());
+                }
+            }
+
+            @Nested
+            @DisplayName("주어진 유저의 정보가 올바르지 않을 때")
+            class Context_without_correct_user_attributes {
+                @Test
+                @DisplayName("bad request 를 응답한다.")
+                void It_respond_bad_request() throws Exception {
+                    mockMvc.perform(
+                            patch("/users/1")
+                                    .contentType(MediaType.APPLICATION_JSON)
+                                    .content("{\"name\":\"\",\"password\":\"\"}")
+                                    .header("Authorization", "Bearer " + VALID_TOKEN)
+                    )
+                            .andExpect(status().isBadRequest());
+                }
+            }
+        }
```

Comment:
이렇게 했다면 컨텍스트의 의미를 더 살릴 수 있었을 거에요.
```suggestion
            @Nested
            @DisplayName("주어진 유저의 정보가 올바르지 않을 때")
            class Context_without_correct_user_attributes {
                final String givenUserData = "{\"name\":\"\",\"password\":\"\"}";

                @Test
                @DisplayName("bad request 를 응답한다.")
                void It_respond_bad_request() throws Exception {
                    mockMvc.perform(
                            patch("/users/1")
                                    .contentType(MediaType.APPLICATION_JSON)
                                    .content(givenUserData)
                                    .header("Authorization", "Bearer " + VALID_TOKEN)
                    )
                            .andExpect(status().isBadRequest());
                }
            }
        }
```